feat: `autoSave` for edit pages (#4591)
refinedev/refine: documentation/docs/api-reference/antd/hooks/form/useDrawerForm.md

// You can use it like this:
{elapsedTime >= 4000 && <div>this takes a bit longer than expected</div>}
```

### `autoSave`

If you want to save the form automatically, you can pass the `autoSave` prop to the this hook. It is useful when you want to save the form automatically when the user changes the form values. 

It also supports `onMutationSuccess` and `onMutationError` callback functions. You can use `isAutoSave` parameter to determine whether the mutation is triggered by `autoSave` or not.

```tsx
useDrawerForm({
    autoSave: true,
})
```

:::caution
Works only in `action: "edit"` mode.
:::

`onMutationSuccess` and `onMutationError` callbacks will be called after the mutation is successful or failed.

### `autoSaveDebounce`

Set the debounce time for the `autosave` prop. Default value is `1000`.

```tsx
useDrawerForm({
    autosave: true,
    // highlight-next-line
    autoSaveDebounce: 2000,

> })

Reviewer: Let's wrap auto-save related props into some object.

// You can use it like this:
{elapsedTime >= 4000 && <div>this takes a bit longer than expected</div>}
```

### `autoSave`

> If you want to save the form automatically after some time when user edits the form, you can pass the `autoSave` prop to the this hook. 

Reviewer: ```suggestion
If you want to save the form automatically after some delay when user edits the form, you can pass true to `autoSave.enabled` prop.
```

console.log(overtime.elapsedTime); // undefined, 1000, 2000, 3000 4000, ...
```

### `autoSaveProps`

> If `autoSave` prop is enabled, the hook returns an `autoSaveProps` object with `data`, `error`, and `status` properties. `data` holds the mutation's returned data, `error` stores any returned error, and `status` indicates the mutation's status.

Reviewer: ```suggestion
If `autoSave` is enabled, this hook returns `autoSaveProps` object with `data`, `error`, and `status` properties from mutation.
```
